components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -467,7 +467,11 @@ def pickFirstSignature(condition, filterLambda):
 
             # Check for Date objects
             # XXXbz Do we need to worry about security wrappers around the Date?
-            pickFirstSignature("%s.get().is_object() && JS_ObjectIsDate(cx, &%s.get().to_object())" %
+            pickFirstSignature("%s.get().is_object() && "
+                               "{ rooted!(in(cx) let obj = %s.get().to_object()); "
+                               "let mut is_date = false; "
+                               "JS_ObjectIsDate(cx, obj.handle(), &mut is_date); "
```

Comment:
Let's assert that JS_ObjectIsDate returns true in order to catch any failures in the JS engine.